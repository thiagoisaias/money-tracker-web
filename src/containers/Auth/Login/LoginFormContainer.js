import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { clearAuthError, login } from "../../../store/actions/auth/auth";
import LoginForm from "../../../components/Auth/LoginForm/LoginForm";

export class LoginFormContainer extends Component {
  submitData = formData => {
    const { history, isLoading, onLogin } = this.props;

    if (isLoading) {
      return;
    }

    onLogin(formData, history);
  };

  render() {
    const { onLogin, isLoading, error } = this.props;
    return (
      <LoginForm
        error={error}
        isLoading={isLoading}
        onLogin={onLogin}
        submitData={this.submitData}
      />
    );
  }

  componentWillUnmount() {
    const { error, onClearAuthError } = this.props;
    if (error) {
      onClearAuthError();
    }
  }
}

LoginFormContainer.propTypes = {
  error: PropTypes.string,
  onClearAuthError: PropTypes.func.isRequired,
  onLogin: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    isLoading: state.auth.isLoading,
    error: state.auth.error
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onClearAuthError: () => {
      dispatch(clearAuthError());
    },

    onLogin: (loginData, history) => {
      dispatch(login(loginData, history));
    }
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(LoginFormContainer)
);
