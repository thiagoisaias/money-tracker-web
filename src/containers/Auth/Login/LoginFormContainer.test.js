import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { LoginFormContainer } from "./LoginFormContainer";

describe("LoginFormContainer", () => {
  const mockProps = {
    error: null,
    onClearAuthError: jest.fn(),
    onLogin: jest.fn(),
    isLoading: false,
    history: {}
  };
  const component = shallow(<LoginFormContainer {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
