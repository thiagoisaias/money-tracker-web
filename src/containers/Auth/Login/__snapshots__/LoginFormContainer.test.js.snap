// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormContainer renders properly 1`] = `
<Component
  error={null}
  isLoading={false}
  onLogin={[Function]}
  onSubmitData={[Function]}
/>
`;
